velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -53,8 +54,77 @@ void WindowFuzzer::addWindowFunctionSignatures(
-  // TODO: allow randomly generating the frame clauses.
-  return "RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW";
+  auto frameType = [](int value) -> const std::string {
+    switch (value) {
+      case 0:
+        return "RANGE";
+      case 1:
+        return "ROWS";
+      default:
+        VELOX_UNREACHABLE("Unknown value for frame type generation");
+    }
+  };
+  auto isRowsFrame =
+      boost::random::uniform_int_distribution<uint32_t>(0, 1)(rng_);
+  auto frameTypeString = frameType(isRowsFrame);
+
+  constexpr int64_t kMax = std::numeric_limits<int64_t>::max();
+  constexpr int64_t kMin = std::numeric_limits<int64_t>::min();
+  auto frameBound =
+      [&](core::WindowNode::BoundType boundType) -> const std::string {
+    // Generating only constant bounded k PRECEDING/FOLLOWING frames for now.
+    auto kValue =
+        boost::random::uniform_int_distribution<int64_t>(kMin, kMax)(rng_);
+    switch (boundType) {
+      case core::WindowNode::BoundType::kUnboundedPreceding:
+        return "UNBOUNDED PRECEDING";
+      case core::WindowNode::BoundType::kPreceding:
+        return fmt::format("{} PRECEDING", kValue);
+      case core::WindowNode::BoundType::kCurrentRow:
+        return "CURRENT ROW";
+      case core::WindowNode::BoundType::kFollowing:
+        return fmt::format("{} FOLLOWING", kValue);
+      case core::WindowNode::BoundType::kUnboundedFollowing:
+        return "UNBOUNDED FOLLOWING";
+      default:
+        VELOX_UNREACHABLE("Unknown option for frame clause generation");
+    }
+  };
+
+  // Generating k PRECEDING and k FOLLOWING frames only for ROWS type.
+  // k RANGE frames require more work as we have to generate columns with the
+  // frame bound values.
+  std::vector<core::WindowNode::BoundType> startBoundOptions, endBoundOptions;
+  if (isRowsFrame) {
+    startBoundOptions = {
+        core::WindowNode::BoundType::kUnboundedPreceding,
+        core::WindowNode::BoundType::kPreceding,
+        core::WindowNode::BoundType::kCurrentRow,
+        core::WindowNode::BoundType::kFollowing};
+    endBoundOptions = {
+        core::WindowNode::BoundType::kPreceding,
+        core::WindowNode::BoundType::kCurrentRow,
+        core::WindowNode::BoundType::kFollowing,
+        core::WindowNode::BoundType::kUnboundedFollowing};
+  } else {
+    startBoundOptions = {
+        core::WindowNode::BoundType::kUnboundedPreceding,
+        core::WindowNode::BoundType::kCurrentRow};
+    endBoundOptions = {
+        core::WindowNode::BoundType::kCurrentRow,
+        core::WindowNode::BoundType::kUnboundedFollowing};
+  }
+
+  // End bound option should not be greater than start bound option as this
+  // would result in an invalid frame.
+  auto startBoundIndex = boost::random::uniform_int_distribution<uint32_t>(
+      0, startBoundOptions.size() - 1)(rng_);
+  auto endBoundIndex = boost::random::uniform_int_distribution<uint32_t>(
+      startBoundIndex, endBoundOptions.size() - 1)(rng_);
```

Comment:
Would this still generate frames like "... between 1 following and 3 following"? This kind of frames is actually helpful as it helped us found a bug before. So it would be nice to be able to generate them.